File: 2z.kt
import java.util.*

fun main() {
    print("Enter a Roman numeral: ")
    val num = readLine()
    if (num != null) {
        println("Number: ${romToInt(num)}")
    }
}

fun romToInt(s: String): Int {
    val romNums = mapOf('I' to 1, 'V' to 5, 'X' to 10, 'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000)
    var res = romNums[s[s.length - 1]] ?: 0
    for (i in s.length - 2 downTo 0) {
        if (romNums[s[i]]!! < romNums[s[i + 1]]!!) {
            res -= romNums[s[i]]!!
        } else {
            res += romNums[s[i]]!!
        }
    }
    return res
}

File: 2z.cs
using System;
using System.Collections.Generic;

class Program
{
    static int RomToInt(string s)
    {
        Dictionary<char, int> romNums = new Dictionary<char, int>
        {
            {'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}
        };

        int res = romNums[s[^1]]; // Численное значение для последнего символа

        for (int i = s.Length - 2; i >= 0; i--)
        {
            if (romNums[s[i]] < romNums[s[i + 1]])
            {
                res -= romNums[s[i]]; // Если меньшая цифра перед большей, вычитаем ее значение
            }
            else
            {
                res += romNums[s[i]]; // Складываем значения
            }
        }

        return res;
    }

    static void Main()
    {
        Console.Write("Enter a Roman numeral: ");
        string num = Console.ReadLine();
        Console.WriteLine("Number: " + RomToInt(num));
    }
}